components/header/wallet_modal.tsx

```diff
@@ -0,0 +1,125 @@
+import { Dispatch, FormEvent, SetStateAction, useState } from 'react';
+import Image from 'next/image';
+import ReactModal from 'react-modal';
+
+import { validateWalletAddress } from '@utils';
+
+import arconnectLogo from '../../public/arconnect.png';
+import {
+	CloseModalButton,
+	ImportWalletErrorMessage,
+	ImportWalletForm,
+	ImportWalletInput,
+	ModalButton,
+	ModalButtons,
+	ModalContent,
+	ModalText,
+} from './wallet_modal.style';
+
+const WalletModal = ({
+	isOpen,
+	closeModal,
+	setWalletAddress,
+}: {
+	isOpen: boolean;
+	closeModal: Function;
+	setWalletAddress: Dispatch<SetStateAction<string>>;
+}): JSX.Element => {
+	const [isArconnectAvailable, setIsArconnectAvailable] = useState(false);
+	const [showWalletInput, setShowWalletInput] = useState(false);
+	const [importedPublicWalletAddress, setImportedPublicWalletAddress] = useState('');
+	const [importedPublicWalletAddressError, setImportedPublicWalletAddressError] = useState('');
+
+	if (typeof window !== 'undefined') {
+		window.addEventListener('arweaveWalletLoaded', async () => {
+			setIsArconnectAvailable(Boolean(window.arweaveWallet));
+
+			try {
+				const walletAddress = await window.arweaveWallet.getActiveAddress();
+				setWalletAddress(walletAddress);
+			} catch (e) {
+				console.log(e);
+			}
+		});
+	}
+
+	const connectToArconnect = async () => {
+		await window.arweaveWallet.connect(['ACCESS_ADDRESS'], { name: 'Inferno Rewards Leaderboard' });
+		const walletAddress = await window.arweaveWallet.getActiveAddress();
+
+		setWalletAddress(walletAddress);
+		closeModal();
+	};
+
+	const onImportWalletSubmit = (e: FormEvent<HTMLFormElement>) => {
+		const data = new FormData(e.target as HTMLFormElement);
+		const walletAddress = data.get('wallet') ? String(data.get('wallet')) : '';
```

Comment:
`data.get()` is probably pretty extremely cheap, but we don't need to call it twice here. Something like this but with better types:

```ts
const walletData: unknown | undefined = data.get('wallet')
const walletAddress = walletData === undefined ? String(walletData) : ''
```

Reply:
https://github.com/ardriveapp/inferno-leaderboard/pull/3/commits/54899614707fdcf86942c972f2c99a1aca5fe13a